crates/core/tedge_agent/src/http_rest.rs

```diff
@@ -0,0 +1,299 @@
+use futures::StreamExt;
+use hyper::{server::conn::AddrIncoming, Request};
+use hyper::{Body, Response, Server};
+use routerify::{Router, RouterService};
+use std::{io::Write, path::Path};
+use std::{net::SocketAddr, path::PathBuf};
+
+#[cfg(not(test))]
+use tedge_config::{ConfigSettingAccessor, TmpPathSetting};
+use tedge_utils::paths::create_directories;
+use tokio::io::AsyncReadExt;
+
+use crate::error::AgentError;
+const FILE_TRANSFER_HTTP_ENDPOINT: &'static str = "/tedge/*";
+const WHAT_TO_NAME_THIS: &'static str = "tedge/file-transfer"; // TODO
+
+#[cfg(not(test))]
+lazy_static::lazy_static! {
+    static ref FILE_TRANSFER_ROOT_PATH: PathBuf = {
+        let config = tedge_config::get_tedge_config().unwrap();  // TODO
+        let value = config.query(TmpPathSetting).unwrap();  // TODO
+        let value: PathBuf = value.into();
+        let root_path = PathBuf::from(format!("{}/{WHAT_TO_NAME_THIS}", value.display()));
+        root_path
+    };
+}
+
+#[cfg(test)]
+lazy_static::lazy_static! {
+    static ref FILE_TRANSFER_ROOT_PATH: PathBuf = {
+        let ttd = tedge_test_utils::fs::TempTedgeDir::new();
+        dbg!(&ttd.path());
+        let root_path = PathBuf::from(format!("{}/{WHAT_TO_NAME_THIS}", ttd.path().display()));
+        root_path
+    };
+}
```

Comment:
This whole concept is not good, sorry. First of all, the root path should not be a `static`. You found yourself that reading from configuration in a static context is not possible (hence your "TODO" markers).

That is a no-go, in my opinion, and needs to be refactored. Passing the root path or the whole configuration to the handlers sounds like the way to go for this.